launcher/modplatform/EnsureMetadataTask.cpp

```diff
@@ -484,10 +495,13 @@ void EnsureMetadataTask::updateMetadata(ModPlatform::IndexedPack& pack, ModPlatf
     try {
         // Prevent file name mismatch
         ver.fileName = resource->fileinfo().fileName();
-        if (ver.fileName.endsWith(".disabled"))
-            ver.fileName.chop(9);
+        QFileInfo ver_info(ver.fileName);
+        if (ver_info.suffix().compare("disabled", Qt::CaseInsensitive) == 0) {
+            ver.fileName.chop(QString(".disabled").size());
```

Comment:
Does the size of a constant change? Or why do we need to compute it?
```suggestion
            ver.fileName.chop(9);
```